File: RepoTransBench/target_projects/Java/Go/NewBeginning6_Subdir-vul-find/public_tests/re_public_test.go
package public_tests

import (
	"testing"
	"newbeginning6subdir/org/example"
)

func TestEscapeAndPatternPublic(t *testing.T) {
	input := "x.y$^"
	escaped := example.ReEscape(input)
	if escaped == "" {
		t.Errorf("ReEscape(%q) returned empty string", input)
	}

	pattern := "\\d+"
	if !example.RePattern(pattern, "2024") {
		t.Errorf("RePattern did not match valid string")
	}
	if example.RePattern(pattern, "test") {
		t.Errorf("RePattern matched invalid string")
	}
	if example.RePattern("abc.*", "xyz") {
		t.Errorf("RePattern matched wrong pattern (abc.* vs xyz)")
	}
}

func TestContainsPublic(t *testing.T) {
	if !example.ReContains("Goodbye moon", "moon") {
		t.Errorf("ReContains did not find substring")
	}
	if example.ReContains("Goodbye", "sun") {
		t.Errorf("ReContains should not find non-existent substring")
	}
	if example.ReContains("", "def") {
		t.Errorf("ReContains should be false for empty haystack")
	}
	if example.ReContains("def", "") {
		t.Errorf("ReContains should be false for empty needle")
	}
	if example.ReContains("", "") {
		t.Errorf("ReContains with both empty should be false")
	}
}

File: RepoTransBench/source_projects/Java/NewBeginning6_Subdir-vul-find/src/test/java/org/example/RePublicTest.java
package org.example;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class RePublicTest {

    @Test
    void testEscapeAndPattern() {
        String input = "x.y$^";
        String escaped = Re.escape(input);
        assertNotNull(escaped);

        String pattern = "\\d+";
        assertTrue(Re.pattern(pattern, "2024"));
        assertFalse(Re.pattern(pattern, "test"));
        assertFalse(Re.pattern("abc.*", "xyz"));
    }

    @Test
    void testContains() {
        assertTrue(Re.contains("Goodbye moon", "moon"));
        assertFalse(Re.contains("Goodbye", "sun"));
        assertFalse(Re.contains(null, "def"));
        assertFalse(Re.contains("def", null));
        assertFalse(Re.contains(null, null));
    }
}